frontend/src/components/session-in-progress/session-controls/session-controls.tsx

```diff
@@ -38,6 +38,7 @@ const SessionControls = (props: SessionControlsProps): ReactElement => {
         payload: newQuestions,
       });
     } else {
+      dispatch({ type: "disable-exit-warning" });
```

Comment:
you're dispatching this at the wrong time. this is getting dispatched whenever a professor hits "next question"